src/masks/InteractionMasks.tsx

```diff
@@ -140,54 +125,83 @@ export default function InteractionMasks<R>({
+        break;
+      case 'ArrowRight':
+        nextPosition = { idx: idx + 1, rowIdx };
+        break;
+      case 'Tab':
+        nextPosition = shiftKey ? { idx: idx - 1, rowIdx } : { idx: idx + 1, rowIdx };
+        break;
       default:
-        return selectedPosition;
+        nextPosition = { idx, rowIdx };
+        break;
     }
+
+    return getNextSelectedCellPosition({
+      columns,
+      rowsCount: rows.length,
+      cellNavigationMode: mode,
+      nextPosition
+    });
   }
 
-  function onKeyDown(e: React.KeyboardEvent<HTMLDivElement>): void {
-    if (isCtrlKeyHeldDown(e)) {
-      onPressKeyWithCtrl(e);
-    } else if (e.keyCode === KeyCodes.Escape) {
-      onPressEscape();
-    } else if (e.keyCode === KeyCodes.Tab) {
-      onPressTab(e);
-    } else if (isKeyboardNavigationEvent(e)) {
-      changeCellFromEvent(e);
-    } else if (isKeyPrintable(e.keyCode) || [KeyCodes.Backspace, KeyCodes.Delete, KeyCodes.Enter].includes(e.keyCode)) {
-      openEditor(e);
+  function onKeyDown(event: React.KeyboardEvent<HTMLDivElement>): void {
+    const { key } = event;
+    if (enableCellCopyPaste && isCtrlKeyHeldDown(event)) {
+      // event.key may be uppercase `C` or `V`
+      const lowerCaseKey = event.key.toLowerCase();
+      if (lowerCaseKey === 'c') return handleCopy();
+      if (lowerCaseKey === 'v') return handlePaste();
     }
-  }
 
-  function openEditor(event: React.KeyboardEvent<HTMLDivElement>): void {
-    if (!isEditorEnabled && isCellEditable(selectedPosition)) {
-      setFirstEditorKeyPress(event.key);
-      setIsEditorEnabled(true);
+    switch (key) {
+      case 'Escape':
+        closeEditor();
+        setCopiedPosition(null);
+        break;
+      case 'Tab':
+        onPressTab(event);
+        break;
+      case 'ArrowUp':
+      case 'ArrowDown':
+      case 'ArrowLeft':
+      case 'ArrowRight':
+        event.preventDefault();
+        selectCell(getNextPosition(key));
+        break;
+      default:
+        break;
     }
-  }
 
-  function closeEditor(): void {
-    setIsEditorEnabled(false);
-    setFirstEditorKeyPress(null);
+    if (canOpenEditor(event)) {
+      setSelectedPosition(({ idx, rowIdx }) => ({ idx, rowIdx, status: 'EDIT', key }));
+    }
   }
 
-  function onPressKeyWithCtrl({ keyCode }: React.KeyboardEvent<HTMLDivElement>): void {
-    if (!enableCellCopyPaste) return;
-    if (keyCode === KeyCodes.c) {
-      handleCopy();
-    } else if (keyCode === KeyCodes.v) {
-      handlePaste();
-    }
+  function canOpenEditor(event: React.KeyboardEvent<HTMLDivElement>): boolean {
+    if (selectedPosition.status === 'EDIT' || !isCellEditable(selectedPosition)) return false;
+
+    const column = columns[selectedPosition.idx];
+    const row = rows[selectedPosition.rowIdx];
+    const isEditable = column.unsafe_onCellInput?.(event, row);
+
+    // Let the user override the activation logic
+    if (typeof isEditable === 'boolean') return isEditable;
```

Comment:
I am not 100% sure about this but I think it makes sense to allow users to override activation logic completely